src/dune_rules/coq/coq_rules.ml

```diff
@@ -570,6 +570,26 @@ let get_dep_map ~dir ~wrapper_name : Path.t list Dep_map.t Action_builder.t =
       ]
 ;;
 
+(* EJGA: Would be nice to have a functor in [Stdune.Tuple] to do this *)
+module DWInput : Memo.Input with type t = Path.Build.t * string = struct
+  type t = Path.Build.t * string
+
+  let hash = Tuple.T2.hash Path.Build.hash String.hash
+  let equal = Tuple.T2.equal Path.Build.equal String.equal
+  let to_dyn = Tuple.T2.to_dyn Path.Build.to_dyn String.to_dyn
+end
+
+(* EJGA: with this code, we have a combined memo node that will be
+   re-executed iff either [dir], [wrapper_name], or the deps of the
+   original action builder (in this case [dep_theory_file ~dir
+   ~wrapper_name]) *)
```

Comment:
Is the word "change" missing from the second-half of the "iff"?  Also, "either" implies two options, so maybe "iff any of [dir], [wrapper_name], or the deps of the original action builder (in this case [dep_theory_file ~dir ~wrapper_name]) change"?

Reply:
Indeed, this was missing, thanks!